### ISSUE TITLE
Use of graphics makes documentation inaccessible to screen reader users 

### ISSUE THREAD

**Issue by Ralcidonis** [home-assistant/home-assistant.github.io#24496]
### Feedback

Please do not use "graphics" or "screenshots" in the body of the documentation to show examples. This is not accessible for users of assistive technology / screen readers. Please use text to give examples. It is still fine to include a screen shot for visual users but this needs to be in addition to the text. Otherwise, the documentation is not helpful to screen reader users.  

### URL

https://www.home-assistant.io/integrations/google/#service-googlecreate_event

### Version

2022/10/01

### Additional information

Thanks.

**Comment by homeassistant** [comment]
Hey there @allenporter, mind taking a look at this feedback as it has been labeled with an integration (`google`) you are listed as a [code owner](https://github.com/home-assistant/home-assistant.io/blob/current/CODEOWNERS#L242) for? Thanks!

**Comment by allenporter** [comment]
Thank you for the report, usable documentation, for everyone, is a priority!

My impression is that the screenshots here are entirely supplemental to the written instructions. Can you help me understand an example of where that is not the case and this can be improved?

**Comment by allenporter** [comment]
Like oauth step 3? It doesn't say "enter a code" but shows a screenshot of code text box.

**Comment by Ralcidonis** [comment]
Hi Allen:

 

Thank you for considering. Indeed, that would be a good example. However, the one that prompted me to submit the feedback is regarding what I believe is an example of how to use the “Google.create_calendar_event” service. My screen reader was reading a graphic right under that entry and I believe it is the actual information that I was looking for.

**Comment by allenporter** [comment]
That particular entry says:

"You can use the service google.create_event to create a new calendar event in a calendar."

Then it has an expandable text box with the title "Create Event Service details" with more information about the service. There is not a screenshot there, but maybe that box is a problem.

I'll do some more investigation.

**Comment by allenporter** [comment]
I installed the "Screen Reader" chrome extension, however I am not sure it represents the right experience.  I am curious if the "Create Event Service details" was the problematic entry given that it is a collapsable detail box. When you click it open, it has a description of the service parameters in a table.

This collapsable detail box is a common widget used in many parts of the documentation, and used a few times in that page, so i'm wondering if I am missing something else that may be the problem.

**Comment by Ralcidonis** [comment]
Indeed, that was the problem. The screen reader is not seeing it as a “collapsible” or “clickable” box leading one to believe it is a graphical element. I interpreted it as a screenshot but this was what I walked-away with.

 

I just went back on the page:

 

The “Create event Service Details” appears as if it is the title of the graphical element that appears right below it.

 

I clicked on the create event details and indeed, the information came to life in a nicely displayed table. I would not have known that this was a “clickable” element outside of this discussion.

 

The same thing for the “more configuration….”

 

Thanks for considering.

**Comment by allenporter** [comment]
Thank you for the extra detail. This is a big problem as there are many, many, clickable elements like throughout the entire documentation on this page and other pages. I will do some research for a fix.

**Comment by Ralcidonis** [comment]
Please do not hesitate to reach out if you would like me to try it again once you have identified a fix.

 

I am thinking, if it is feasible, that inserting a screen reader flag may be the best option as to not avoid modifying existing work. Only the screen reader will see the flag identifying the element as “clickable.”

**Comment by allenporter** [comment]
I did some reading about the attributes to add to give more semantic context to the screen reader. I installed a reader so I think I have what I need for now, but may ask for you to re-review later once the docs are updated.  Thanks.

**Comment by Ralcidonis** [comment]
Glad to  be part of the community! Thanks.

**Comment by allenporter** [comment]
@Ralcidonis the [documentation](https://www.home-assistant.io/integrations/google/) has been updated, including many other pages with the same trouble. I tested this on two different readers, but would like your review as well when you have a moment. Thanks!

**Comment by Ralcidonis** [comment]
Beautiful! It is now fairly intuitive to the screen reader by adding the collapsible element. Thanks for the quick fix.

**Comment by allenporter** [comment]
Awesome, thank you for the report.
Marking as closed.
